### PR TITLE
Migrate Jenkinsfile to GH Actions Workflow

### PR DESCRIPTION
**Pull request by reakaleek**
## Summary
Migrate Jenkinsfile to GH Actions Workflow

## Tasks
- [x] Migrate Jenkinsfile
- [x] Delete Jenkins Job config
- [ ] Delete Jenkins Job
- [ ] Update sheet

### CODE REVIEW

**Comment by reakaleek** [comment]
Hi @trentm!

I need your help here. I'm currently migrating the Jenkinsfile to a GH Actions Workflow.

As you can see in the status checks, the test for `node 17` is failing, but only in GH Actions. The Jenkins stage is still successful. The main difference seems to be that Jenkins is utilising docker, while GH Actions seems to be using a native node version. 

While I was trying to debug the issue, I found that both actions/setup-node@v3 and docker are using `node 17.9.1`.

Interestingly, I was able to reproduce the same behaviour locally. When running the install and test script in the `node:17` docker container, the tests succeeded.

But running it with `node 17` while utilising `nvm`, the tests are failing the same way as is in the GH Action workflow.

I will take any hint, advice, opinion, suggestion or help!

Thank you.

cc @cachedout

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [10, 12, 14, 16, 17]
```
Comment (reakaleek) [inline .github/workflows/test.yml:22]: Also @trentm, what is your opinion on expanding this list?

We could add `18` and `19`. WDYT?

Comment (trentm) [reply .github/workflows/test.yml:22]: Yes we could change that to `[10, 12, 14, 16, 18, 19]`.

Comment (reakaleek) [inline .github/workflows/test.yml:22]:
```suggestion
        node-version: [10, 12, 14, 16, 18, 19]
```

Comment (reakaleek) [reply .github/workflows/test.yml]: https://github.com/elastic/ecs-logging-nodejs/pull/132/commits/03752ef936c169f3a421e1603d5abd23a36a1713

Oh, no. It's also failing for 18 and 19.

**Comment by trentm** [comment]
@reakaleek Can you cherry-pick this commit to your branch? https://github.com/elastic/ecs-logging-nodejs/commit/afcba058271c0c345684619b0600b8d4e1dde31e   This should fix the tests for node >=v17.

**Comment by reakaleek** [comment]
> @reakaleek Can you cherry-pick this commit to your branch? [afcba05](https://github.com/elastic/ecs-logging-nodejs/commit/afcba058271c0c345684619b0600b8d4e1dde31e) This should fix the tests for node >=v17.

HI @trentm,

Thank you for the help. Unfortunately, the test for for node 19 is now failing with a different error. It seems to be related to https://nodejs.org/en/blog/announcements/v19-release-announce/#http-s-1-1-keepalive-by-default

**Comment by reakaleek** [comment]
@trentm I was able to make the tests green with https://github.com/elastic/ecs-logging-nodejs/pull/132/commits/54426d985613c50ae72dcaaed376f65b076afbb6.

Let me know what you think and if this patch is sufficient for your use case. Otherwise, please propose a change request.

**Review by kuisathaverat — APPROVED** [review]

File: helpers/test/basic.test.js [hunk 2]
```diff
@@ -122,6 +122,7 @@ test('formatHttpRequest and formatHttpResponse should return a valid ecs object'
         })
       }
     )
+    req.setHeader('connection', 'close')
```
Comment (trentm) [inline helpers/test/basic.test.js:125]: Perhaps instead add `connection: 'close'` to the `headers: { ... }` object above on line 110?

**Review by trentm — APPROVED** [review]
Looks great. Thanks!

**Review by cachedout — APPROVED** [review]
@reakaleek This looks fine but I am curious why you made the changes from `localhost` -> `127.0.0.1`? Was this just for syntactic consistency or was there a functional reason for these changes?

**Comment by reakaleek** [comment]
> @reakaleek This looks fine but I am curious why you made the changes from `localhost` -> `127.0.0.1`? Was this just for syntactic consistency or was there a functional reason for these changes?

This change was actually cherry-picked by the suggestion here: https://github.com/elastic/ecs-logging-nodejs/pull/132#issuecomment-1362182198

File: helpers/test/basic.test.js (hunk 2 above)
Comment (reakaleek) [reply helpers/test/basic.test.js]: https://github.com/elastic/ecs-logging-nodejs/pull/132/commits/44da905e31f900162b8471916c4006b7edd46a53

**Pull request merged**